src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -297,6 +298,74 @@ public AgentSettings LoadSettings()
             }
         }
 
+        public async Task UnconfigureAsync(CommandSettings command)
+        {
+            if (!_store.IsConfigured())
+            {
+                _term.WriteError(StringUtil.Loc("MissingSettings"));
+            }
+
+            if (!_store.HasCredentials())
+            {
+                _term.WriteError(StringUtil.Loc("MissingCredentials"));
+            }
+
+            //stop, uninstall service and remove service config file
+            if (_store.IsServiceConfigured())
+            {
+                var serviceControlManager = HostContext.GetService<IServiceControlManager>();
+                try
+                {
+                    serviceControlManager.UnconfigureService();
+                    _term.WriteLine(StringUtil.Loc("ServiceUninstalled"));
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error(ex);
+                    _term.WriteError(StringUtil.Loc("ServiceUninstallFailed"));
+                }
+            }
+
+            //delete agent from the server
+            try
+            {
+                AgentSettings settings = _store.GetSettings();
+                var credentialManager = HostContext.GetService<ICredentialManager>();
+                VssCredentials creds = credentialManager.LoadCredentials();
+                Uri uri = new Uri(settings.ServerUrl);
+                VssConnection conn = ApiUtil.CreateConnection(uri, creds);
+                var agentSvr = HostContext.GetService<IAgentServer>();
+                await agentSvr.ConnectAsync(conn);
+                await agentSvr.DeleteAgentAsync(settings.PoolId, settings.AgentId);
+                _term.WriteLine(StringUtil.Loc("AgentDeleted"));
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+                _term.WriteError(StringUtil.Loc("DeleteAgentFailed"));
+            }
+
+            //delete credential config file
+            try
+            {
+                _store.DeleteCredential();
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+            }
+
+            //delete settings config file
+            try
+            {
+                _store.DeleteSettings();
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+            }
+        }
```

Comment:
I'll re-throw as suggested. I'll have to create AggregateException, because we don't want to skip any steps right?
Even that exceptions are swallowed, most of the catch handlers are printing to the console to the user so it is not a silent failure. The only catch handlers that do not throw are for DeleteSettings and DeleteCredential, but this is because I figured the only reason these fail is if files are missing and I have put special check for that case in the beginning of the function (which prints to stderrr).